deployments/definition_store.go

```diff
@@ -1129,11 +1140,10 @@ func createNodeInstances(consulStore consulutil.ConsulStore, kv *api.KV, numberI
 	}
 }
 
-/**
-This function check if a nodes need a floating IP, and return the name of Floating IP node.
-*/
-func checkFloattingIP(kv *api.KV, deploymentID, nodeName string) (bool, string, error) {
-	requirementsKey, err := GetRequirementsKeysByTypeForNode(kv, deploymentID, nodeName, "network")
+// This function check if a nodes need a specified requirement with a specified capacity type or derived
+// and return the name of the related node addressing the capacity
+func checkRequirements(kv *api.KV, deploymentID, nodeName, requirement, capabilityType string) (bool, string, error) {
```

Comment:
I think it is an interesting function that could be exposed as part of the `deployments` package api. We need to change it's name, move it to the requirements file and document it a bit more. WDYT?

Reply:
Yes, this function seems to be very closed to the HasAnyRequirement in the prov terraform commons package you mentioned below